uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -299,14 +299,14 @@ private static function savePublicHolidaysBalanceChanges(
   ) {
     $balanceChangeTypes = array_flip(LeaveBalanceChange::buildOptions('type_id', 'validate'));
 
-    $publicHolidays = $calculation->getPublicHolidaysInEntitlement();
+    $publicHolidays = $calculation->getNumberOfPublicHolidaysInEntitlement();
 
     if (!empty($publicHolidays)) {
       LeaveBalanceChange::create([
         'type_id'     => $balanceChangeTypes['public_holiday'],
         'source_id'   => $periodEntitlement->id,
         'source_type' => LeaveBalanceChange::SOURCE_ENTITLEMENT,
-        'amount'      => count($publicHolidays)
+        'amount'      => $publicHolidays
       ]);
```

Comment:
Please format the array parameters passed to the create method properly. With single space between the array `=>` operator and the array key and values.